### PR TITLE
Added test coverage for BZ#1863597 and BZ#1920072

### PR DESCRIPTION
**Pull request by devendra104**
In this PR, We are checking the inventory cloud plugin and puppet server tuning parameter update using the satellite installer.

**Test Results:**

```
$ pytest -s tests/foreman/installer/test_installer.py::test_installer_inventory_plugin_update 
tests/foreman/installer/test_installer.py
.
================================================================================================ warnings summary =================================================================================================
=================================================================================== 1 passed, 3 warnings in 1633.30s (0:27:13) ====================================================================================

$ pytest -s tests/foreman/installer/test_installer.py::test_installer_puppet_overload
plugins: reportportal-5.0.8, xdist-2.3.0, services-2.2.1, ibutsu-1.16, forked-1.3.0, mock-3.6.1
collecting ... 2021-08-31 14:23:18 - robottelo.collection - INFO - Processing test items to add testimony token markers
collected 1 item            
tests/foreman/installer/test_installer.py 
.
================================================================================================ warnings summary =================================================================================================
=================================================================================== 1 passed, 3 warnings in 1647.20s (0:27:27) ====================================================================================

```

### CODE REVIEW

File: tests/foreman/installer/test_installer.py [hunk 1]
```diff
@@ -1584,3 +1585,79 @@ def test_installer_timestamp_logs():
...
+    :CaseImportance: Medium
+
+    :CaseLevel: System
+
+    :BZ: 1863597
+
+    :customerscenario: true
+
+    """
+    ContentHost(destructive_sat.hostname).create_custom_repos(rhel7=settings.repos.rhel7_repo)
```
Comment (omkarkhatavkar) [inline tests/foreman/installer/test_installer.py:1610]: `destructive_sat` is inherited from the ContentHost would rather have the `create_custom_repos` by default.
`
 destructive_sat.create_custom_repos(rhel7=settings.repos.rhel7_repo)` is not working for you ?

File: tests/foreman/installer/test_installer.py [hunk 2]
```diff
@@ -1584,3 +1585,79 @@ def test_installer_timestamp_logs():
 
     :CaseAutomation: NotAutomated
     """
+
+
+def test_installer_inventory_plugin_update(destructive_sat):
```
Comment (omkarkhatavkar) [inline tests/foreman/installer/test_installer.py:1590]: Do you think we need to add `@pytest.mark.destructive` any ways currently everything will be running un-grouped manner, having that marker would help in collection.

**Review by omkarkhatavkar — APPROVED** [review]
ACK, pending some queries.

File: tests/foreman/installer/test_installer.py (hunk 1 above)
Comment (devendra104) [reply tests/foreman/installer/test_installer.py:1610]: yes, you are right, it works.

File: tests/foreman/installer/test_installer.py [hunk 3]
```diff
@@ -1663,3 +1663,81 @@ def test_positive_capsule_installer_logfile_check():
...
+    :CaseLevel: System
+
+    :BZ: 1920072
+
+    :customerscenario: true
+    """
+    puppet_server_jvm_param_1 = '-Djruby.logger.class=com.puppetlabs.jruby_utils.jruby.Slf4jLogger'
+    puppet_server_jvm_param_2 = '-XX:ReservedCodeCacheSize=1024m'
+    command_output = destructive_sat.execute(
+        f'satellite-installer --puppet-server-jvm-extra-args={puppet_server_jvm_param_1}'
```
Comment (jameerpathan111) [inline tests/foreman/installer/test_installer.py:1731]: @devendra104 There is `InstallerCommand` helper in `robottelo/helpers.py` to construct satellite-installer commands, do you think we can use it in this test?

Comment (devendra104) [reply tests/foreman/installer/test_installer.py:1731]: Thanks, @jameerpathan111  It's a nice suggestion.

**Review by jameerpathan111 — APPROVED** [review]
LGTM

**Comment by devendra104** [comment]
**Test Results:**
```
$pytest -s tests/foreman/installer/test_installer.py::test_installer_inventory_plugin_update
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.0, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/desingh/Satellite-QE/Robottelo/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.8, xdist-2.3.0, services-2.2.1, ibutsu-1.16, forked-1.3.0, mock-3.6.1

tests/foreman/installer/test_installer.py [INFO 210902 23:58:53] Using provider AnsibleTower to checkout
.
================================================================================================ warnings summary =================================================================================================
=================================================================================== 1 passed, 3 warnings in 1849.25s (0:30:49) ====================================================================================

$ pytest -s tests/foreman/installer/test_installer.py::test_installer_puppet_overload
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.0, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/desingh/Satellite-QE/Robottelo/robottelo, configfile: pyproject.toml
collected 1 item                                                                                                                                                                                                  

tests/foreman/installer/test_installer.py [INFO 210903 11:07:04] Using provider AnsibleTower to checkout
.


================================================================================================ warnings summary =================================================================================================
=================================================================================== 1 passed, 3 warnings in 1919.16s (0:31:59) ====================================================================================
```

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Pull request merged**